tests/Integration/Behaviour/Features/Scenario/Order/order_currency.feature

```diff
@@ -0,0 +1,115 @@
+      | total_shipping_tax_incl  | 74.20  |
+
+#  @database-scenario
+  @reboot-kernel-before-scenario
+  Scenario: Add cart rule of type 'amount' to an order with secondary currency
+    Given I add discount to order "bo_order1" with following details:
+      | name      | discount ten-euros    |
+      | type      | amount                |
+      | value     | 10                    |
+    Then order "bo_order1" should have 1 cart rule
+    Then order "bo_order1" should have cart rule "discount ten-euros" with amount "€9.43"
+    Then order "bo_order1" should have following details:
+      | total_products           | 238.00 |
+      | total_products_wt        | 252.28 |
+      | total_discounts_tax_excl | 9.43   |
+      | total_discounts_tax_incl | 10.0   |
+      | total_paid_tax_excl      | 298.57 |
+      | total_paid_tax_incl      | 316.48 |
+      | total_paid               | 316.48 |
+      | total_paid_real          | 0.0    |
+      | total_shipping_tax_excl  | 70.00  |
+      | total_shipping_tax_incl  | 74.20  |
+
+#  @database-scenario
+  @reboot-kernel-before-scenario
+  Scenario: Add cart rule of type 'percent' to an order with secondary currency
+    Given I add discount to order "bo_order1" with following details:
+      | name      | discount ten-percents |
+      | type      | percent               |
+      | value     | 10                    |
+    Then order "bo_order1" should have 1 cart rule
+    Then order "bo_order1" should have cart rule "discount ten-percents" with amount "€23.80"
+    Then order "bo_order1" should have following details:
+      | total_products           | 238.00 |
+      | total_products_wt        | 252.28 |
+      | total_discounts_tax_excl | 23.80  |
+      | total_discounts_tax_incl | 25.23  |
+      | total_paid_tax_excl      | 284.20 |
+      | total_paid_tax_incl      | 301.25 |
+      | total_paid               | 301.25 |
+      | total_paid_real          | 0.0    |
+      | total_shipping_tax_excl  | 70.00  |
+      | total_shipping_tax_incl  | 74.20  |
+
+#  @database-scenario
+  @reboot-kernel-before-scenario
+  Scenario: Add product to an order with secondary currency
+    When I add products to order "bo_order1" without invoice and the following products details:
+      | name          | Mug Today is a good day  |
+      | amount        | 5                      |
+      | price         | 15                      |
+    Then order "bo_order1" should have 0 cart rule
+    Then order "bo_order1" should have following details:
+      | total_products           | 313.00 |
+      | total_products_wt        | 331.78 |
+      | total_discounts_tax_excl | 0.00   |
+      | total_discounts_tax_incl | 0.00   |
+      | total_paid_tax_excl      | 383.00 |
+      | total_paid_tax_incl      | 405.98 |
+      | total_paid               | 405.98 |
+      | total_paid_real          | 0.0    |
+      | total_shipping_tax_excl  | 70.00  |
+      | total_shipping_tax_incl  | 74.20  |
```

Comment:
Nice scenarios!

Missing EOL here however
And could you add some additional tests please:
- add product
- edit product
- free shipping discount
- generate invoice and check the details (with a discount)
- change Carrier to check shipping is correctly computed

I know it is not the scope of the issue but since we don't have tests in other currencies it would be safer to ensure it works correctly rather than finding out later

Most of these tests already exist and just need to be copied here with adapted values